datafusion/expr/src/udf.rs

```diff
@@ -467,7 +478,28 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     /// to arrays, which will likely be simpler code, but be slower.
     ///
     /// [invoke_no_args]: ScalarUDFImpl::invoke_no_args
-    fn invoke(&self, _args: &[ColumnarValue]) -> Result<ColumnarValue>;
+    fn invoke(&self, _args: &[ColumnarValue]) -> Result<ColumnarValue> {
+        not_impl_err!(
+            "Function {} does not implement invoke but called",
+            self.name()
+        )
+    }
+
+    /// Invoke the function with `args` and the number of rows,
+    /// returning the appropriate result.
+    ///
+    /// The function should be used for signatures with [`datafusion_expr_common::signature::Volatility::Volatile`]
+    /// and with arguments.
```

Comment:
Only for these?

if yes => the function should be named appropriate `invoke_volatile`)
if no => remove the comment